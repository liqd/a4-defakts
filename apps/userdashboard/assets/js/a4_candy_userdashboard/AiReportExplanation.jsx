import React, { useState } from 'react'
import django from 'django'
import { SwitchButton } from '../../../../../apps/contrib/assets/SwitchButton'

const translated = {
  intro: django.pgettext('defakts', 'This comment contains disinformation. Defakts uses an Artificial ' +
  'Intelligence to scan content for disinformation. Disinformation often shows ' +
  'certain characteristics that allow for a reliable identification.'),
  ariaReadMore: django.pgettext('defakts', 'Click to view the AI explanation for reporting this comment.'),
  ariaReadLess: django.pgettext('defakts', 'Click to hide the AI explanation for reporting this comment.'),
  readMore: django.pgettext('defakts', 'Read more'),
  showLess: django.pgettext('defakts', 'Show less'),
  showInfoSwitch: django.pgettext('defakts', 'Show info to users'),
  hideInfoSwitch: django.pgettext('defakts', 'Hide info from users')
}

export const AiReportExplanation = ({ AiReport, notificationPk }) => {
  const [isExpanded, setIsExpanded] = useState()

  const toggleExpand = () => {
    setIsExpanded(!isExpanded)
  }

  const toggleReadMore = (
    <button
      className="btn--link text-danger"
      type="button"
      onClick={toggleExpand}
      aria-label={isExpanded ? translated.ariaReadLess : translated.ariaReadMore}
    >
      {isExpanded ? translated.showLess : translated.readMore}
    </button>
  )

  return (
    <li className="alert alert--danger mb-4">
      <div className="d-flex text-start mb-4">
        <i
          className="fas fa-exclamation-circle text-danger pt-1 pe-2"
          aria-hidden="True"
        />

        {!isExpanded
          ? (
            <p className="pe-4">{translated.intro} {toggleReadMore}</p>
            )
          : (
            <div className="pe-4">
              <p>{translated.intro}</p>
              <p>{AiReport} {toggleReadMore}</p>
            </div>
            )}
      </div>
      <div className="d-flex text-start">
        <SwitchButton
          initiallyChecked
          id={notificationPk}
          switchLabelOn={translated.hideInfoSwitch}
          switchLabelOff={translated.showInfoSwitch}
        />
      </div>
    </li>
  )
}
