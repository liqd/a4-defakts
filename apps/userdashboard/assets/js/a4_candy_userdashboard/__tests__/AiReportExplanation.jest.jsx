import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import { AiReportExplanation } from '../AiReportExplanation'

test('Snapshot Read More <AiReportExplanation>', () => {
  const { container } = render(
    <AiReportExplanation AiReport="This is the ai report" />
  )
  expect(container).toMatchSnapshot()
})

test('Snapshot Read Less <AiReportExplanation>', () => {
  const { container } = render(
    <AiReportExplanation AiReport="This is the ai report" />
  )
  const button = screen.getByRole('button')
  fireEvent.click(button)
  expect(container).toMatchSnapshot()
})

test('Render <AiReportExplanation>', () => {
  render(<AiReportExplanation AiReport="This is the ai report" />)
  const comment = screen.getByText('Read more')
  expect(comment).toBeTruthy()
})

test('Read More <AiReportExplanation>', () => {
  render(<AiReportExplanation AiReport="This is the ai report" />)
  const readMore = screen.getByText('Read more')
  expect(readMore).toBeTruthy()
  const button = screen.getByRole('button')
  expect(button).toBeTruthy()
  fireEvent.click(button)
  const readLess = screen.getByText('Read less')
  expect(readLess).toBeTruthy()
})
