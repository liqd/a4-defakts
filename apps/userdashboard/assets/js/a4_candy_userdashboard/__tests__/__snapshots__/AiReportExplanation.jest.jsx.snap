// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot to confirm Read Less is rendered after clicking Read More button <AiReportExplanation> 1`] = `
<div>
  <li
    class="alert alert--danger mb-4"
  >
    <div
      class="d-flex text-start fs-6 mb-4"
    >
      <i
        aria-hidden="True"
        class="fas fa-exclamation-circle text-danger pt-1 pe-2"
      />
      <div
        class="pe-4"
      >
        <p>
          This comment contains disinformation. Defakts uses an Artificial Intelligence to scan content for disinformation. Disinformation often shows certain characteristics that allow for a reliable identification.
        </p>
        <p>
          This is the ai report
           
          <button
            aria-label="Click to hide the AI explanation for reporting this comment."
            class="btn--link text-danger"
            type="button"
          >
            Show less
          </button>
        </p>
      </div>
    </div>
    <div
      class="d-flex text-start"
    >
      <button
        aria-checked="true"
        aria-labelledby="switch-id-undefined"
        class="btn-switch"
        role="switch"
      >
        <span
          class="btn-switch__switch"
        >
          <span
            class="btn-switch__adjuster"
          />
        </span>
        <span
          id="switch-id-undefined"
        >
          Hide info from users
        </span>
      </button>
    </div>
  </li>
</div>
`;

exports[`Snapshot to confirm Read More is rendered <AiReportExplanation> 1`] = `
<div>
  <li
    class="alert alert--danger mb-4"
  >
    <div
      class="d-flex text-start fs-6 mb-4"
    >
      <i
        aria-hidden="True"
        class="fas fa-exclamation-circle text-danger pt-1 pe-2"
      />
      <p
        class="pe-4"
      >
        This comment contains disinformation. Defakts uses an Artificial Intelligence to scan content for disinformation. Disinformation often shows certain characteristics that allow for a reliable identification.
         
        <button
          aria-label="Click to view the AI explanation for reporting this comment."
          class="btn--link text-danger"
          type="button"
        >
          Read more
        </button>
      </p>
    </div>
    <div
      class="d-flex text-start"
    >
      <button
        aria-checked="true"
        aria-labelledby="switch-id-undefined"
        class="btn-switch"
        role="switch"
      >
        <span
          class="btn-switch__switch"
        >
          <span
            class="btn-switch__adjuster"
          />
        </span>
        <span
          id="switch-id-undefined"
        >
          Hide info from users
        </span>
      </button>
    </div>
  </li>
</div>
`;
